server/src/main/java/org/tctalent/server/api/admin/UserAdminApi.java

```diff
@@ -135,12 +136,26 @@ public void sendVerifyEmailRequest(@RequestBody SendVerifyEmailRequest request)
     }
 
     @GetMapping("/verify-email/{token}")
-    public void verifyEmail(@PathVariable("token") String token) throws ExpiredEmailTokenException, InvalidEmailVerificationTokenException {
-        VerifyEmailRequest request = new VerifyEmailRequest();
-        request.setToken(token);
-        userService.verifyEmail(request);
+    public ModelAndView verifyEmail(@PathVariable("token") String token) {
+        ModelAndView modelAndView = new ModelAndView();
+        try {
+            VerifyEmailRequest request = new VerifyEmailRequest();
+            request.setToken(token);
+            User user = userService.verifyEmail(request);
```

Comment:
Instead of returning the user by changing verifyEmail to return the user, I would expose a UserService method which extracts a user from a token.

Then you can extract the User before doing the verify.

That means you can use the User to pass the real displayName in case of exceptions instead of having to insert "User".